### ISSUE TITLE
Using HA app

### ISSUE THREAD

**Issue by TobiasPSP** [brenner-tobias/addon-cloudflared#677]
### The problem

It may just be a documentation bug - I followed all instructions, and the tunnel works like a charm. Kudos for your hard work.

I was under the impression that I would be able to use this tunnel with HA apps as well. I am currently using the HA app for iPhone.

However, the app does not connect, there is no meaningful error message, and in the docs and elsewhere I could neither find mention of extra setup steps required to enable apps, or a clear statement that apps aren't supported and the tunnel is limited to normal browsers.

I'd greatly appreciate if you could point me into the right direction. 

your add-on is so good and helpful that I am currently setting up instructions for noobs like me that are *really* simple to follow, and I'd like to include a definite statement in respect to using apps - any authoritative feedback appreciated.

### What version of Cloudflared has the issue?

5.1.15

### What type of installation are you running?

Home Assistant OS

### Steps to reproduce the issue

On an iphone not connected to WiFi, in the browser navigate to the tunnel domain name = works perfectly well.
When opening the Home Assistant app, the screen basically freezes.

**Comment by elcajon** [comment]
Hey @TobiasPSP, the (HA companion) apps should work out of the box with the Cloudflare tunnel.
There are known problems if [application rules](https://developers.cloudflare.com/cloudflare-one/policies/access/) are defined for the Cloudflare tunnel in Cloudflares zero trust dashboard (e.g. https://github.com/brenner-tobias/addon-cloudflared/issues/654#issuecomment-2162698784).

You should check if the external URL (which can be configured in [HA network settings](https://my.home-assistant.io/redirect/network/)) is correct and points to your Cloudflare tunnel URL. This setting is used to configure the external URL used by the companion apps while setting up the App for the first time.
Otherwise I would assume that there is a DNS cache problem with the iPhone or the iOS HA app. You may need to clear the cache in the app settings or reinstall the app and set it up from scratch.

**Comment by TobiasPSP** [comment]
Thank you very much for your prompt reply. Greatly appreciated.

I am using no Zero Trust (yet) and followed strictly the guide (local tunnel, all tunnel settings from add-in, no additional hardening except mandating https).

Maybe you would like to look over the steps I took which I published here: https://done.land/tools/software/homeassistant/internetaccess

Ultimate goal is to have a one-stop guide so this is going to expand for Zero Trust and many other secondary options.

If you come across errors in my write-up please correct me. I am fairly new to most of this.

Where in HA do I specify the external address?

**Comment by elcajon** [comment]
> If you come across errors in my write-up please correct me. I am fairly new to most of this.

Looks pretty good to me, really good work!
As there are no problems accessing HA through your browser, I wouldn't expect any problems related to the Cloudflare tunnel.

> Where in HA do I specify the external address?

[Settings > System > Network > Home Assistant URL](https://my.home-assistant.io/redirect/network/)

<img width="626" alt="image" src="https://github.com/user-attachments/assets/7d0b7a5a-b19e-4b3a-b85d-e29d27da7965">

If you don't see this option, you need to enable 'advanced mode' for your user.

[Profile Picture (bottom left) > General](https://my.home-assistant.io/redirect/profile/) > User settings > Toggle Advanced mode

**Comment by TobiasPSP** [comment]
Thank you.
I had set this setting already. I just reinstalled the app but no difference:

when I launch the app, it shows a white screen and an orange gear icon in its right lower corner. It then stays like this forever. Web access in the browser works like a charm.

I am more than happy with just the browser but for my tutorial (and to understand what’s happening) I‘d love to solve this final part.

Do you maybe have another idea what I could try?

I‘ll meanwhile go ahead and try on my wifes‘ smartphone which never ran any ha stuff before to rule out caching issues.

**Comment by TobiasPSP** [comment]
Ok, so when I install the app on a new fresh smartphone, all runs perfectly, including the app.

which raises the question how I could refresh settings on a smartphone that I used with HA before setting up the tunnel :-)

When I uninstalled the HA app, it claimed to also delete all data associated with it. However, when I reinstalled it, it did not ask me about my HA server and apparently still used old information.

on the „fresh“ smartphone, during installation I went through a couple more setting dialogs…

**Comment by TobiasPSP** [comment]
This is what I get on the original smartphone where the HA app was installed before I added the tunnel:

NSURLErrorDomain code -1001
Request timed out

**Comment by TobiasPSP** [comment]
Ok, here is what solved it: on the problematic smartphone, when I launched the HA app, at the blank screen I clicked the gear icon, then went to servers. There was one entry for the local instance, and another for the domain. I deleted both.

On next app start, the app now behaved like on the „fresh“ smartphone, newly detected the local instance, and when I tried with WiFi off, it automatically connected to the tunnel.

no idea what to make of this, but it seems like a quick way to refresh all connections.

Let me iterate how awesome your add-on is - thanks to your great work, accessing HA from anywhere in a safe way is now possible even for noobs like me 👍
